Classes/Utility/Task/ImportTaskUtility.php

```diff
@@ -102,15 +102,23 @@ public function run($configurationsUids) {
 
                         break;
                     case Token::INSTAGRAM_OAUTH2:
-                        //getting data array from instagram api json result
-                        $url = sprintf('https://api.instagram.com/v1/users/%s/media/recent/?access_token=%s&count=%d',
-                            $configuration->getSocialId(),
-                            $configuration->getToken()->getCredential('accessToken'),
-                            $configuration->getFeedsLimit()
-                        );
-
+                        // getting data array from instagram api json result
+						// in case of hashtag
+						if (substr($configuration->getSocialId(), 0, strlen('#')) === '#') {
+                            $url = sprintf('https://api.instagram.com/v1/tags/%s/media/recent/?access_token=%s&count=%d',
```

Comment:
This same string appears twice, replace it with either a variable or a setting: 'https://api.instagram.com/v1/tags/%s/media/recent/?access_token=%s&count=%d'